units/pod_allocator.go

```diff
@@ -169,37 +180,89 @@ func (j *podAllocatorJob) populate() error {
 			return errors.Wrapf(err, "finding task '%s'", j.TaskID)
 		}
 		if t == nil {
-			return errors.New("task not found")
+			return errors.Errorf("task '%s' not found", j.TaskID)
 		}
 		j.task = t
 	}
 
+	if j.pRef == nil {
```

Comment:
Yeah, the nil check is for testing. It'll only be nil if you roundtrip the job through the remote queue due to serialization. Calling `j.Run` directly or using a local queue to run it will preserve `j.pRef` if it's set.